### ISSUE TITLE
Review PromptModal logic

### ISSUE THREAD

**Issue by cnotv** [rancher/dashboard#6085]
**Detailed Description**
Review data and logic related to the `PromptModal`.

**Context**
For the use of modals we have a generic action extended to several components. The data is too much generic and there's not defined common actions as separated key/value. 
This generates confusion, bloated data and also issues like [in this case](https://github.com/rancher/dashboard/pull/5887#discussion_r887061526)

**Comment by catherineluse** [comment]
For the list of receivers for AlertmanagerConfigs in project monitoring, I ran into this problem too. For me the issue was not just that it was a generic action, but that it was tightly coupled to the Resource class, so it was hard to make it work on a list view of something that isn't a Kubernetes resource.

I modified the component to take props, so that its show/hide state could be controlled with props instead of the generic piece of state in Vuex, which was also tightly coupled with data from the Resource class. See the comments I left on the new props: https://github.com/rancher/dashboard/blob/master/shell/components/ActionMenu.vue#L31

If we refactored the list views to control the state with props and use a single ActionMenu component on each list view, instead of just using a single ActionMenu throughout the entire application, I think that it would make the code a lot more readable and maintainable.
